boards/saml21-xpro: add Arduino API support for saml21-xpro board

fedepell opened the pull request:
### Contribution description
Adds the Arduino API support for the Atmel SAML21-XPRO board.
This enables Arduino-like programming supported by RIOT also on this board.

### Testing procedure
Arduino automatic test sys_arduino can be used for basic functionalities tests:

```
cd tests/sys_arduino/
BOARD=saml21-xpro make all flash test
```

Additionally one can manually use and play with the examples/arduino_hello-world/ for other tests (ie. playing with I/O and leds)

PeterKietzmann commented:
I'm not too familiar with the Arduino wrappers but don't we usually provide a "arduino_pinmap.h" file to create pin Arduino<->RIOT mappings? @aabadie?

fedepell commented:
@PeterKietzmann : thanks for the feedback! Indeed I checked more examples and saw that most of them have the files split into two .h as you suggest. So I modified my PR splitting into two and squashed together.
Hope it looks better and more in line with other boards now!

dylad commented:
Changes look good but I'm wondering what are the benefits from adding arduino support for SAML21 ? Is there any arduino board featuring this MCU ? or using the same header connectors ?

fedepell commented:
In my case (don't know if this can be generalized) the main point was to enable some internal developers (more hw oriented) with Arduino knowledge and already existing code to develop on this board.

fedepell commented:
@dylad : it wasn't the scope of my PR but now I noticed that indeed there is a SAM arduino-like board indeed out there:
https://startingelectronics.org/articles/arduino/wemos-arduino-m0/
So maybe it would be more interesting in this aspect. I will try to get my hands on the board and add support for it too.

dylad commented:
@fedepell the board you link here is samd21-based so this won't help with this PR.
The only saml21-based board that I know is saml21-xpro from Atmel but there is no arduino connectors on it so I am really puzzle with this PR.

fedepell commented:
@dylad: no worries about this PR! If you think my use case could not be useful to others and the project no worries just close it, I just made the PR thinking it may be of use to someone else as well :) I can manage this small thing in our local repository!

PeterKietzmann commented:
@dylad we don't only support the arduino API for boards with arduino pinout. For example, we have esp32 or stm32disco boards implementing it.

PeterKietzmann commented:
I did a basic code review and set the related labels. Unfortunately I'm not able to provide any testing here

dylad commented:
@PeterKietzmann Good to know ! I'll handle tests this week so this PR can finally be merged.

dylad commented on boards/saml21-xpro/include/arduino_pinmap.h:
```diff
@@ -0,0 +1,46 @@
+
+/**
+ * @ingroup     boards_saml21-xpro
+ * @brief       Mappinf from board pins to Arduino pins
```
Mapping

fedepell replied:
fixed now, thanks!

dylad commented:
Please fix typo before we merge :) you can squash directly.

fedepell commented:
Thanks for checking! Just corrected the typo and squashed!

dylad commented:
All green -> Let's go.
